src/overlays/actors/ovl_Bg_Ydan_Sp/z_bg_ydan_sp.c

```diff
@@ -63,29 +73,377 @@ static ColliderTrisInit D_808C09A8 = {
+
+        tri[1].x = thisx->world.pos.x + (cossY * ti1->dim.vtx[0].x) - (ti1->dim.vtx[2].y * sinsY * nSinsX);
+        tri[1].y = thisx->world.pos.y + (ti1->dim.vtx[2].y * cossX);
+        tri[1].z = thisx->world.pos.z - (sinsY * ti1->dim.vtx[0].x) + (ti1->dim.vtx[2].y * cossY * nSinsX);
+        Collider_SetTrisVertices(&this->trisCollider, 1, &tri[0], &tri[2], &tri[1]);
+    }
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+    this->timer = 0;
+    if (Flags_GetSwitch(globalCtx, this->isDestroyedSwitchFlag)) {
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void BgYdanSp_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgYdanSp* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(globalCtx, &this->trisCollider);
+}
+
+void* BgYdanSp_UpdateFloorWebCollision(BgYdanSp* this) {
+    s16 newY;
+    CollisionHeader* colHeader;
+
+    colHeader = (CollisionHeader*)SEGMENTED_TO_VIRTUAL(&D_06006460);
+    colHeader->vtxList = (Vec3s*)SEGMENTED_TO_VIRTUAL(colHeader->vtxList);
+    newY = (s16)((this->dyna.actor.home.pos.y - this->dyna.actor.world.pos.y) * 10.0f);
+    colHeader->vtxList[14].y = newY;
+    colHeader->vtxList[12].y = newY;
+    colHeader->vtxList[10].y = newY;
+    colHeader->vtxList[9].y = newY;
+    colHeader->vtxList[6].y = newY;
+    colHeader->vtxList[5].y = newY;
+    colHeader->vtxList[1].y = newY;
+    colHeader->vtxList[0].y = newY;
+}
+
+void BgYdanSp_BurnWeb(BgYdanSp* this, GlobalContext* globalCtx) {
+    this->timer = 30;
+    this = this;
+    func_80078884(NA_SE_SY_CORRECT_CHIME);
+    Flags_SetSwitch(globalCtx, this->isDestroyedSwitchFlag);
+    if (this->dyna.actor.params == WEB_FLOOR) {
+        this->actionFunc = &BgYdanSp_BurnFloorWeb;
+        return;
+    }
+    this->actionFunc = &BgYdanSp_BurnWallWeb;
+}
+
+void BgYdanSp_BurnFloorWeb(BgYdanSp* this, GlobalContext* globalCtx) {
+    static Vec3f accel = { 0 };
+    Vec3f velocity;
+    Vec3f pos2;
+    f32 distXZ;
+    f32 sins;
+    f32 coss;
+    s16 rot;
+    s16 rot2;
+    s32 i;
+
+    DECR(this->timer);
+
+    if (this->timer == 0) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    if ((this->timer % 3) == 0) {
+        rot2 = (s16)(Rand_ZeroOne() * 0x2AAA);
+        velocity.y = 0.0f;
+        pos2.y = this->dyna.actor.world.pos.y;
+
+        for (i = 0; i < 6; i++) {
+            rot = (s16)(Rand_CenteredFloat(0x2800) + rot2);
+            sins = Math_SinS(rot);
+            coss = Math_CosS(rot);
+            pos2.x = this->dyna.actor.world.pos.x + (120.0f * sins);
+            pos2.z = this->dyna.actor.world.pos.z + (120.0f * coss);
+            distXZ = Math_Vec3f_DistXZ(&this->dyna.actor.home.pos, &pos2) * (1.0f / 120.0f);
+            if (distXZ < 0.7f) {
+                sins = Math_SinS((s16)(rot + 0x8000));
+                coss = Math_CosS((s16)(rot + 0x8000));
```

Comment:
I would expect a lot of the s16 casts in this file to not be necessary, but I could be wrong